### PR TITLE
mobile: removing the duplicate autonomous upstream

### PR DESCRIPTION
**Pull request by alyssawilk**
The Envoy test upstream is more flexible than the E-M test upstream and removing the duplicate class with similar functionality will hopefully avoid future confusion

Risk Level: n/a 
Testing:
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

**Review by RyanTheOptimist — APPROVED** [review]
Thank so much for doing this cleanup!

**Review by jpsim — DISMISSED** [review]

**Review by jpsim — DISMISSED** [review]

File: test/integration/fake_upstream.h [hunk 1]
```diff
@@ -88,7 +88,7 @@ class FakeStream : public Http::RequestDecoder,
   void encodeHeaders(const Http::HeaderMap& headers, bool end_stream);
   void encodeData(uint64_t size, bool end_stream);
   void encodeData(Buffer::Instance& data, bool end_stream);
-  void encodeData(absl::string_view data, bool end_stream);
+  void encodeData(std::string data, bool end_stream);
```
Comment (adisuissa) [inline test/integration/fake_upstream.h:91]: nit: can `data` be either const-ref or string_view?

File: test/integration/autonomous_upstream.h [hunk 2]
```diff
@@ -91,9 +91,11 @@ class AutonomousUpstream : public FakeUpstream {
   void setLastRequestHeaders(const Http::HeaderMap& headers);
   std::unique_ptr<Http::TestRequestHeaderMapImpl> lastRequestHeaders();
   void setResponseTrailers(std::unique_ptr<Http::TestResponseTrailerMapImpl>&& response_trailers);
+  void setResponseBody(std::string body);
```
Comment (adisuissa) [inline test/integration/autonomous_upstream.h:94]: nit: can `body` be either const-ref or string_view?

File: test/integration/autonomous_upstream.cc [hunk 3]
```diff
@@ -77,9 +77,13 @@ void AutonomousStream::sendResponse() {
...
   encodeHeaders(upstream_.responseHeaders(), headers_only_response);
   if (!headers_only_response) {
-    for (int32_t i = 0; i < response_data_blocks; ++i) {
-      encodeData(response_body_length,
-                 i == (response_data_blocks - 1) && !send_trailers && end_stream);
+    if (upstream_.responseBody().has_value()) {
+      encodeData(*upstream_.responseBody(), !send_trailers);
+    } else {
+      for (int32_t i = 0; i < response_data_blocks; ++i) {
+        encodeData(response_body_length,
```
Comment (adisuissa) [inline test/integration/autonomous_upstream.cc:84]: Should `headers_only_response` (line 71) also take the `upstream_.responseBody().has_value()` value into account?

**Review by adisuissa — COMMENTED** [review]
Thanks.
Left a few nits on the non-mobile part.

**Comment by adisuissa** [comment]
Assigning @jpsim for the E-M code changes.

**Review by jpsim — APPROVED** [review]
+1 on the mobile side of things

File: test/integration/autonomous_upstream.cc (hunk 3 above)
Comment (alyssawilk) [reply test/integration/autonomous_upstream.cc:84]: I think if you set headers-only response and set a body you're just doing it wrong =P

**Review by adisuissa — APPROVED** [review]
Discussed with @alyssawilk offline, and decided to leave the nits for future PRs.

**Pull request merged**